bundle-src/src/dashboard/index.select-event.tsx

```diff
@@ -0,0 +1,111 @@
+import "regenerator-runtime";
+import { useEffect, useMemo, useRef, useState } from "react";
+import ReactDOM from "react-dom";
+import { useOnlyReplicantValue, useReplicantValue } from "common/useReplicant";
+import {
+  Box,
+  Button,
+  Center,
+  Container,
+  Group,
+  Loader,
+  MantineProvider,
+  Paper,
+  Stack,
+  Title,
+} from "@mantine/core";
+import type { BaseEventType } from "@ystv/scores/src/common/types";
+import { EventID } from "common/types/eventID";
+import invariant from "invariant";
+
+function Dashboard() {
+  const [events, setEvents] = useState<BaseEventType[] | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  useEffect(() => {
+    async function get() {
+      setError(null);
+      try {
+        const result = await nodecg.sendMessage("list-events");
+        setEvents(result);
+      } catch (e) {
+        console.error("list-events ERROR", e);
+        if (e instanceof Error) {
+          setError(e.message);
+        } else {
+          setError(JSON.stringify(e));
+        }
+      }
+    }
+    get();
+  }, []);
+
+  const [selected, setSelected] = useState<string | null>(null);
+  useEffect(() => {
+    nodecg.readReplicant<EventID>("eventID", (val) => {
+      if (selected === null && val !== null) {
+        setSelected(val.replace(/Event\/.*\/(.+)$/, "$1"));
+      }
+    });
+  }, []);
+
+  const [_, setRepValue] = useReplicantValue<EventID>("eventID");
+
+  useEffect(() => {
+    const listener = () => {
+      console.log("Confirmed!", selected);
+      // The API expects the ID to be in the form `Event/<type>/<id>`
+      invariant(events, "got dialog-confirm before events were loaded");
+      const selectedEvt = events.find((x) => x.id === selected);
+      invariant(selectedEvt, "did not find event for selection");
+      setRepValue(`Event/${selectedEvt.type}/${selectedEvt.id}`);
+    };
+    document.addEventListener("dialog-confirmed", listener);
+    return () => document.removeEventListener("dialog-confirmed", listener);
+  }, [selected]);
+
+  return (
+    <div style={{ margin: "1rem 4rem" }}>
+      <MantineProvider theme={{}}>
```

Comment:
normally I wouldn't like multiple mantine providers in a single project, but given the implementation I guess it makes sense to work inside the nodecg ecosystem!

Reply:
In theory we _could_ share the theme between graphics-src and scores-src, though I wonder whether we'd want to have different themes for each given the different target audiences (graphics ops on PCs vs. producers on phones)

Reply:
I meant more there is one provider for the dialog and one for the main dashboard, coz I would've assumed it would be a built-in modal/react router type situation, but because of the architecture of this it's fine I think. Definitely fine with having a different provider for scores interface and graphics interface lol

Reply:
Yeah I'm not terribly pleased with NodeCG's dialog system so might rip this out for a modal at some point - the main downside would be that it's stuck inside the dashboard iframe, though this is a fullbleed panel so it's not as bad.